CFC_DataCollector/recommender/recommendation_pipeline.py

```diff
@@ -0,0 +1,15 @@
+def get_trips_to_improve(user_id, filter_function):
+    #pick trips which we would like to improve
+    #will make usage of canonical trip class
+    #returns a list of trips implementing Trip interface
+    return []
+
+def retrieve_alternative_trips(trip_list):
```

Comment:
Also, where is the perturbation? Is it going to be hidden in the interface? If so, you might want to document it here just to avoid confusion.